charts/argo-templates/templates/diff-bq-table.yaml

```
@@ -137,6 +137,20 @@ spec:
           - name: rows-to-append-count
             valueFrom:
               parameter: '{{ "{{tasks.query-row-count.outputs.result}}" }}'
+      metrics:
+        prometheus:
+          {{- include "argo.collect-execution-metrics" . | indent 10 }}
+          - name: rows-to-delete-gauge
```

I think this is an invalid prometheus metric name (because of the hyphens). The suffix is also redundant with the type. I'd suggest:
```suggestion
          - name: rows_to_delete
```
And same for other metrics